src/test/isolation2/sql/dependency.sql

```diff
@@ -0,0 +1,801 @@
+-- Test cases when a parallel transaction drops a dependency object
+-- while current transaction is yet not committed.
+
+-- Case 1. Function dependency on the schema.
+create schema test_1_schema;
+
+1: begin;
+1: create function test_1_schema.test_1_function() returns text as $$
+    select 'test'::text; /**/
+$$ language sql;
+
+-- Check that we didn't add extra locks. Here we lock only namespace, so the count is 1.
+-- Do this check only for the first couple of tests in order not to overcomplicate the remaining.
+1:  with cte as (select * from pg_locks where pid = pg_backend_pid() and locktype = 'object')
+select count(1), gp_segment_id from pg_locks
+where classid in (select classid from cte) and objid in (select objid from cte)
+group by gp_segment_id order by gp_segment_id;
```

Comment:
too hard!
why not simply
```suggestion
1: select l.gp_segment_id, count(1)
from pg_locks l
join pg_locks c using (locktype, classid, objid)
where c.pid = pg_backend_pid() and c.locktype = 'object'
group by 1 order by 1;
```